app/controllers/links_controller.rb

```diff
@@ -0,0 +1,29 @@
+class LinksController < ApplicationController
+
+  def new
+    if !session[:user_id].present?
```

Comment:
Same comment as the last PR about using `current_user` instead of digging through the `session` everywhere.